test/e2e/mcd_test.go

```diff
@@ -425,6 +425,46 @@ func TestReconcileAfterBadMC(t *testing.T) {
 	}
 }
 
+func TestFullyDeleteMCShippedFiles(t *testing.T) {
+	cs := framework.NewClientSet("")
+	nodes, err := getNodesByRole(cs, "worker")
+	require.Nil(t, err)
+	for _, node := range nodes {
+		// this fails if the file is on the node, just a pre-req sanity check to run this test
+		execCmdOnNode(t, cs, node, "test", "-e", "/rootfs/etc/something")
+	}
+	workerOldMcFirst := getMcName(t, cs, "worker")
+	mcHostFileFirst := createMCToAddFile("modify-host-file-first", "/etc/something", "mco-test", "root")
+	_, err = cs.MachineConfigs().Create(mcHostFileFirst)
+	require.Nil(t, err)
+	renderedConfig, err := waitForRenderedConfig(t, cs, "worker", mcHostFileFirst.Name)
+	require.Nil(t, err)
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", renderedConfig))
+	workerOldMcSecond := getMcName(t, cs, "worker")
+	mcHostFileSecond := createMCToAddFile("modify-host-file-second", "/etc/something", "mco-test-2", "root")
+	_, err = cs.MachineConfigs().Create(mcHostFileSecond)
+	require.Nil(t, err)
+	renderedConfig, err = waitForRenderedConfig(t, cs, "worker", mcHostFileSecond.Name)
+	require.Nil(t, err)
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", renderedConfig))
+	// now delete the MCs file, which must result in the file disappearing from the system
+	require.Nil(t, cs.MachineConfigs().Delete(mcHostFileSecond.Name, &metav1.DeleteOptions{}))
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", workerOldMcSecond))
+	require.Nil(t, cs.MachineConfigs().Delete(mcHostFileFirst.Name, &metav1.DeleteOptions{}))
+	require.Nil(t, waitForPoolComplete(t, cs, "worker", workerOldMcFirst))
+	nodes, err = getNodesByRole(cs, "worker")
+	require.Nil(t, err)
+	for _, node := range nodes {
+		assert.Equal(t, node.Annotations[constants.CurrentMachineConfigAnnotationKey], workerOldMcFirst)
+		assert.Equal(t, node.Annotations[constants.MachineConfigDaemonStateAnnotationKey], constants.MachineConfigDaemonStateDone)
+		// this fails if the file is on the node
+		_, err := execCmdOnNodeWithErr(t, cs, node, "test", "-e", "/rootfs/etc/something")
```

Comment:
Sorry, don't mean to drag this out. :)  Can we make this `test ! -e /rootfs/etc/something` instead? That way we can check for a successful exit code. Otherwise it's hard to tell if there's an error because the command exited nonzero or something else went wrong along the way in `oc rsh`.

Reply:
done and yeah, definitely better!